scripts/build/utils/generate-types.ts

```diff
@@ -0,0 +1,95 @@
+import { spawn } from 'child_process';
+import limit from 'p-limit';
+import { join, relative } from 'pathe';
+import picocolors from 'picocolors';
+
+import type { BuildEntries } from './entry-utils';
+import { modifyCoreThemeTypes } from './modify-core-theme-types';
+
+const DIR_CODE = join(import.meta.dirname, '..', '..', '..', 'code');
+
+export async function generateTypesFiles(cwd: string, data: BuildEntries) {
+  const DIR_CWD = cwd;
+  const DIR_REL = relative(DIR_CODE, DIR_CWD);
+
+  const dtsEntries = Object.values(data.entries)
+    .flat()
+    .filter((entry) => entry.dts !== false)
+    .map((e) => e.entryPoint);
+
+  // Spawn each entry in it's own separate process, because they are slow & synchronous
+  // ...this way we do not bog down the main process/esbuild and can run them in parallel
+  // we limit the number of concurrent processes to 3, because we don't want to overload the host machine
+  // by trial and error, 3 seems to be the sweet spot between perf and consistency
+  const limited = limit(10);
+  let processes: ReturnType<typeof spawn>[] = [];
+
+  await Promise.all(
+    dtsEntries.map(async (entryPoint) => {
+      return limited(async () => {
+        let timer: ReturnType<typeof setTimeout> | undefined;
+        const dtsProcess = spawn(
+          join(import.meta.dirname, '..', '..', 'node_modules', '.bin', 'jiti'),
+          [join(import.meta.dirname, 'dts-process.ts'), entryPoint],
```

Comment:
This appears not to work on windows. 

```suggestion
          'node',
           [
             join(import.meta.dirname, '..', '..', 'node_modules', 'jiti', 'lib', 'jiti-cli.mjs'),
             join(import.meta.dirname, 'dts-process.ts'),
             entryPoint
           ],
```

would be a bit more portable?

Reply:
I fixed it in a slightly different way here: https://github.com/storybookjs/storybook/pull/32116

thank you ❤️